### PR TITLE
Wildcards

### PR DESCRIPTION
**Pull request by theshane**
Hello, sorry about the confusion. This is my first time contributing to a project like this.

It does not pass perl5.10 but the code works well. Perhaps you can take a look at the tests? I renamed them .proposed so they aren't ran automatically.

### CODE REVIEW

**Comment by jhthorsen** [comment]
Have a look at #52. Send me an email or ask in the PR if you ever consider making a new PR. This is not confusing. It's just not a required thing to do. I also don't mind of the tests are failing. I will have a look anyway, and let you know what I think about the failing tests. (However, failing tests is a blocker for merging)

**Comment by jhthorsen** [comment]
Another trick:

   git rebase -i 0899fcc^
   # an editor opens...
   # add add "s" (for squash) at the similar commits
   # save and exit
   # force push to update this pull request
   git push origin -f master

Again: Look at #52 for more tips, regarding making a new branch.

File: lib/Mojolicious/Plugin/AssetPack.pm [hunk 1]
```diff
@@ -82,6 +83,7 @@ sub register {
   $self->out_dir($self->_build_out_dir($config, $app));
   $self->base_url($config->{base_url}) if $config->{base_url};
   $self->_reloader($app, $config->{reloader}) if $config->{reloader};
+  $self->{static_paths} = $app->static->paths;
```
Comment (jhthorsen) [inline lib/Mojolicious/Plugin/AssetPack.pm]: This is no good. ```$app->static->paths``` can be changed between loading AssetPack and defining assets, so we can't cache the value here.

File: lib/Mojolicious/Plugin/AssetPack.pm [hunk 2]
```diff
@@ -27,7 +28,7 @@ has _ua  => sub {
 
 sub add {
   my ($self, $moniker, @files) = @_;
-
+  @files = $self->_check_for_wildcards(\@files) if(grep(/\*\./, @files));
```
Comment (jhthorsen) [inline lib/Mojolicious/Plugin/AssetPack.pm]: You don't need to check for wildcards here _and_ inside _check_for_wildcards(). Just do it inside _check_for_wildcards(). I would also just pass inn @files and not the ref. If you want to pass in \@files. Or maybe do this instead:

    @files = map { $self->_glob($_) } @files;

File: lib/Mojolicious/Plugin/AssetPack.pm [hunk 3]
```diff
@@ -102,6 +104,49 @@ sub register {
   );
 }
 
+sub _check_for_wildcards {
+    my ( $self, $files ) = @_;
+
+
+    #Keep track of files that we have seen already
+    my %seenFiles;
```
Comment (jhthorsen) [inline lib/Mojolicious/Plugin/AssetPack.pm]: Nice! I think I would probably forget about this... However, should we warn() if we see the same file twice..?

File: lib/Mojolicious/Plugin/AssetPack.pm [hunk 4]
```diff
@@ -102,6 +104,49 @@ sub register {
...
+    #We will replace the files array with this one
+    my @new_files_array;
+    #Loop through files
+    foreach my $file ( @{$files} ) {
```
Comment (jhthorsen) [inline lib/Mojolicious/Plugin/AssetPack.pm]: Not a big deal, but replace ```foreach``` with ```for```. There's no reason what-so-ever to type ```foreach```.

File: lib/Mojolicious/Plugin/AssetPack.pm [hunk 5]
```diff
@@ -102,6 +104,49 @@ sub register {
...
+
+        #Check for *. files EX: *.js, *.css
```
Comment (jhthorsen) [inline lib/Mojolicious/Plugin/AssetPack.pm]: I wonder if this whole thing can be replaced with:

    @files = grep { !$seen{$_}++ } glob $file;

File: lib/Mojolicious/Plugin/AssetPack/Manual/Assets.pod [hunk 6]
```diff
@@ -63,6 +63,27 @@ the current L<Mojolicious> application.
...
+      "/js/myapp.js",
+      "/js/*.js",
+    )
+  );
+
+The /js/*.js will tell AssetPack to look in all public directories for any .js files 
+that are not in the list already. 
+
+If there is a need to set a load precedence, simply add the files that need to be loaded first 
+into the array before the wildcard
```
Comment (jhthorsen) [inline lib/Mojolicious/Plugin/AssetPack/Manual/Assets.pod]: Awesome! I like it :+1:

Should we also note that the files are sorted, so you can do 00-file.js...99-file.js? (Yeah, we need to add sort to the glob logic as well)

File: t/get_wildcard.t.proposed [hunk 7]
```diff
@@ -0,0 +1,28 @@
+use t::Helper;
```
Comment (jhthorsen) [inline t/get_wildcard.t.proposed]: Why is it called proposed? I would rather have a failing test, than a test that is not run.

File: t/js.t [hunk 8]
```diff
@@ -7,6 +7,7 @@ use t::Helper;
 
   $t->app->asset('app.js' => '/js/a.js', '/js/b.js');
 
+
```
Comment (jhthorsen) [inline t/js.t]: Huh? Remove this diff.

**Comment by jhthorsen** [comment]
See #54.

**Comment by theshane** [comment]
I hope this is right. Sorry for polluting the PR list.

**Comment by jhthorsen** [comment]
Closed by faadbd8 and 5b817db.

Let me know if glob() doesn't work as you expect.

**Pull request closed without merging**